downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/BiologicalProcessAnnotationBuilder.java

```diff
@@ -0,0 +1,122 @@
+                    processProteins(proteinInstances, reactionInst);
+                }
+            }
+        }
+    }
+
+    // Returns all protein instances from multi-instance PhysicalEntitys (Polymer/Complex/EntitySet) or just the single incoming instance if it is an EWAS.
+    private static Set<GKInstance> getBiologicalProcessProteins(GKInstance physicalEntityInst) throws Exception {
+        Set<GKInstance> proteinInstances = new HashSet<>();
+        if (GOAGeneratorUtilities.multiInstancePhysicalEntity(physicalEntityInst.getSchemClass())) {
+            proteinInstances.addAll(getMultiInstanceSubInstances(physicalEntityInst));
+        } else if (physicalEntityInst.getSchemClass().isa(ReactomeJavaConstants.EntityWithAccessionedSequence)) {
+            proteinInstances.add(physicalEntityInst);
+        }
+        return proteinInstances;
+    }
+
+    // Attempt to generate GO Annotation information for each protein associated with the whole Reaction or just it's catalysts, depending on the goTerm being evaluated.
+    private static void processProteins(Set<GKInstance> proteinInstances, GKInstance reactionInst) throws Exception {
+
+        for (GKInstance proteinInst : proteinInstances) {
+            GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+            GKInstance speciesInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.species);
+            // Check if the protein has any disqualifying attributes.
+            boolean validProtein = GOAGeneratorUtilities.validateProtein(referenceEntityInst, speciesInst);
+            if (validProtein) {
+                String taxonIdentifier = ((GKInstance) speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+                // TODO: Rule check
+                if (!microbialSpeciesToExclude.contains(taxonIdentifier)) {
+                    getGOBiologicalProcessLine(referenceEntityInst, reactionInst, taxonIdentifier);
+                }
+            }
+        }
+    }
+
+    // Before creating GOA line for BP annotations, the reaction in question needs to be checked for the existance of a 'goBiologicalProcess' attribute. If there is none
+    // than the instance's 'hasEvent' referrals are checked for any.
+    private static List<String> getGOBiologicalProcessLine(GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        List<String> goAnnotationLines = new ArrayList<>();
+        for (Map<String, String> biologicalProcessAccession : getGOBiologicalProcessAccessions(reactionInst, 0)) {
+            String goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, "P", biologicalProcessAccession.get("accession"), biologicalProcessAccession.get("event"), "TAS", taxonIdentifier);
```

Comment:
Can the Strings here be replaced with constants?  Does ReactomeJavaConstants have values for "accession" and "event"?

Reply:
`ReactomeJavaConstants` _does_ have those strings, but I don't think they are being used in the context of Reactome data from the Relational database, so I'd suggest a different set of constants here, _if_ constants are to be used.

Reply:
Yea both those constants exist but Solomon is correct about the context -- here they are being used as keys to a structure I created, as opposed to anything retrieved from the database

Reply:
Done